stream_alert_cli/test.py

```diff
@@ -48,338 +56,486 @@ class TestingSuppressFilter(logging.Filter):
+            [generator] yields a tuple containig a boolean of test status and
+                a list of alerts to run through the alert processor
+        """
+        all_tests_passed = True
+
+        for rule_file, rule_name in get_rule_test_files(rules):
+            with open(os.path.join(DIR_RULES, rule_file), 'r') as rule_file_handle:
+                try:
+                    contents = json.load(rule_file_handle)
+                except Exception as err:
+                    all_tests_passed = False
+                    message = 'Improperly formatted file - {}: {}'.format(
+                        type(err).__name__, err)
+                    self.rules_fail_pass_warn[2].append((rule_name, message))
+                    continue
 
-    # Run the rule processor. Passing 'None' for context will load a mocked object later
-    alerts = StreamAlert(None, True).run(event)
+            test_records = contents.get('records')
+            if not test_records:
+                all_tests_passed = False
+                self.rules_fail_pass_warn[2].append(
+                    (rule_name, 'No records to test in file'))
+                continue
 
-    # Stop mocked sns
-    BOTO_MOCKER_SNS.stop()
+            print_header = True
+            # Go over the records and test the applicable rule
+            for test_record in test_records:
+                if not self.check_keys(rule_name, test_record):
+                    all_tests_passed = False
+                    continue
 
-    # we only want alerts for the specific rule passed in
-    matched_alert_count = len([x for x in alerts if x['metadata']['rule_name'] == rule_name])
+                self.apply_helpers(test_record)
 
-    report_output([test_record['service'], test_record['description']],
-                  matched_alert_count != expected_alert_count)
+                # Run tests on the formatted record
+                alerts, expected_alerts = self.test_rule(
+                    rule_name,
+                    test_record,
+                    self.format_record(test_record))
 
-    return matched_alert_count == expected_alert_count
+                current_test_passed = len(alerts) == expected_alerts
 
-def format_record(test_record):
-    """Create a properly formatted Kinesis, S3, or SNS record.
+                # Print rule name for section header, but only if we get
+                # to a point where there is a record to actually be tested.
+                # This avoids potentialy blank sections
+                if print_header:
+                    if alerts or self.print_output:
+                        print '\n{}'.format(rule_name)
+                        print_header = not print_header
+
+                if self.print_output:
+                    report_output([
+                        current_test_passed,
+                        'rule',
+                        test_record['service'],
+                        '{} [trigger={}]'.format(test_record['description'],
```

Comment:
What item are you referring to? The `[trigger=?]` count? It isn't really added as a column (and I don't want it to be) since it would conflict with alert processor output column. Can you maybe paste an example?